### PR TITLE
Pull module additional images automatically

### PR DESCRIPTION
**Pull request by DavidePrincipi**
- Basic agent actions. Each agent inherits a set of core-defined actions. It is possible to extend and override this basic actions set as wanted.
- New `org.nethserver.images` label. The core `add-module` action inspects the module image labels and extracts the list of additional images. That list is passed to `create-module` input and the basic 10pullimages step downloads the additional images and set the module environment variables.

Environment variables names are set as follow:
- one variable for each image
- variable name is the uppercase value of the image name
- symbols are mapped to `_` (underscore)
- `_IMAGE` suffix is appended

The variable value is the image URL itself.

Example:
- `docker.io/library/mysql:10.3-alpine` becomes `MYSQL_IMAGE=docker.io/library/mysql:10.3-alpine`

### CODE REVIEW

**Review by gsanchietti — COMMENTED** [review]
Why we need to define a new dir named `/usr/local/nethserver/agent/actions`?
Can we use the existing one named ${AGENT_INSTALL_DIR}?

File: core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+agent.set_weight('05pullimages', '5')
+
+for image_url in images:
+    # Extract the image name from the last / to the last :
+    # For instance ghcr.io/nethserver/core:latest => core
+    image_repotag = image_url[image_url.rfind('/')+1:]
+    if ':' in image_repotag:
+        image_name = image_repotag[:image_repotag.rfind(':')]
+    else:
+        image_name = image_repotag
```
Comment (gsanchietti) [inline core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages:42]: May I suggest a more concise version? :)

```suggestion
    url, sep, tag = image_url.parition(':')
    image_name = os.path.basename(url)
```
```

File: core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+import agent
```
Comment (gsanchietti) [inline core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages:23]: This is the for below change :grin: 
```suggestion
import os.path
import agent
```

Comment (DavidePrincipi) [reply core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages:23]: It is not a filesystem path. I'd prefer to not import that module :innocent:

**Comment by DavidePrincipi** [comment]
> Why we need to define a new dir named `/usr/local/nethserver/agent/actions`?
> Can we use the existing one named ${AGENT_INSTALL_DIR}?

It is an _additional_ path owned by the core, where we can put action definitions that are inherited by any agent.

It is picked up by the rootless agents here: https://github.com/NethServer/ns8-scratchpad/pull/45/files#diff-063c7f2c11b942efc86cd3db64cebb8bf419b21781167bd849bef515cfeb8d20R12

File: core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages (hunk 2 above)
Comment (gsanchietti) [reply core/imageroot/usr/local/nethserver/agent/actions/create-module/05pullimages:23]: :sob:

**Review by gsanchietti — APPROVED** [review]

**Review by Amygos — APPROVED** [review]

**Comment by DavidePrincipi** [comment]
Merged manually

**Pull request closed without merging**